include/dlaf/util_math.h

```diff
@@ -41,48 +44,81 @@ constexpr auto ceilDiv(const IntType num, const IntType den)
 }
 #endif
 
-namespace size_t {
 namespace internal {
 
-/// @brief Perform the given binary operation on integer types in size_t arithmetic
+/// @brief Perform the given binary operation on integer types in @tparam ArithmeticT arithmetic
 ///
-/// It casts operands to size_t type and then performs the given binary operation @p op
+/// It casts operands to @tparam ArithmeticT type and then performs the given binary operation @p op
 /// @tparam TA and @tparam TB must be an integer types (signed or unsigned)
 /// @return @p a @p op @p b
-/// @pre @p a >=0 and @p b >= 0
+/// @pre @p a and @p b can be stored in the type @tparam ArithmeticT
+/// @pre it must be possible to store the result in @tparam ArithemticT
 #ifdef DLAF_DOXYGEN
-template <typename TA, typename TB, typename BinaryOp>
+template <class ArithmeticT, class TA, class TB, class BinaryOp>
 constexpr std::size_t generic_integer_op(const TA a, const TB b, BinaryOp op);
```

Comment:
This should be updated as well, (or removed if the enable_if is integrated in the template...)

Reply:
Good catch!

I opted for removing the `std::enable_if`, since the check about types is already performed on both inputs and output type by using `integral_cast`. In this way also the documentation has been unified.